build.sh

```diff
@@ -0,0 +1,161 @@
+     exit 1
+  fi
+}
+
+################################################################################
+function printInfo {
+################################################################################
+echo ---------------------------------------------------------------------------
+echo $*
+echo ---------------------------------------------------------------------------
+}
+
+################################################################################
+# END DECLARATION FUNCTIONS / START OF SCRIPT
+################################################################################
+
+################################################################################
+# Test for commands used in this script
+################################################################################
+testForCommands="dirname git date java javac"
+
+for command in $testForCommands
+do 
+  $command --help >> /dev/null
+  if [ $? -ne 0 ]; then
+    echo "Error: command '$command' is not installed!"
+    exit 1
+  fi
+done
+
+################################################################################
+# Determine directory of script. 
+################################################################################
+ACTUAL_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+################################################################################
+# Check parameters
+################################################################################
+checkParameters $*
+
+################################################################################
+# Determine repo name 
+################################################################################
+REPO_URL=`git remote get-url origin`
+REPO_NAME=${REPO_URL##*/}
+
+printInfo Build microservice of $REPO_NAME at $INSTALLATION_DIRECTORY
+
+
+################################################################################
+# Build service
+################################################################################
+
+echo Build service...
+./gradlew -Prelease clean build
+
+
+echo Copy configuration to $INSTALLATION_DIRECTORY...
+find . -name application-default.properties -exec cp '{}' $INSTALLATION_DIRECTORY/application.properties \;
+
+echo Copy jar file to $INSTALLATION_DIRECTORY...
+find . -name "$REPO_NAME*.jar" -exec cp '{}' $INSTALLATION_DIRECTORY \;
+
+echo Create config directory
+mkdir $INSTALLATION_DIRECTORY/config
+
+echo Create lib directory
+mkdir $INSTALLATION_DIRECTORY/lib
+
+###############################################################################
+# Create run script
+################################################################################
+printInfo Create run script ...
+
+# Determine name of jar file.
+for file in $INSTALLATION_DIRECTORY/*.jar; do
+  jarFile=${file##*/}
+done
+
+echo #!/bin/bash                                                                                >  $INSTALLATION_DIRECTORY/run.sh
```

Comment:
Every line beginning with "echo #" creates just an empty line in the output of build.sh. You need to escape: # -> \# or maybe just make ordinary script comments that will not be transferred into the run.sh. Not sure what is better.

Reply:
echo "...." solves the problem